client/client.go

```diff
@@ -440,19 +447,19 @@ func (c Client) GetTargetsFile(roleName string, keyIDs []string, snapshotMeta da
 
 // RoleTargetsPath generates the appropriate filename for the targets file,
 // based on whether the repo is marked as consistent.
-func (c Client) RoleTargetsPath(roleName string, hashSha256 string, consistent bool) (string, error) {
+func (c Client) RoleTargetsPath(role string, hashSha256 string, consistent bool) (string, error) {
 	if consistent {
-		dir := filepath.Dir(roleName)
-		if strings.Contains(roleName, "/") {
-			lastSlashIdx := strings.LastIndex(roleName, "/")
-			roleName = roleName[lastSlashIdx+1:]
+		dir := filepath.Dir(role)
+		if strings.Contains(role, "/") {
```

Comment:
These paths are TUF paths which only use / regardless of OS. The path is also being build to be used in an HTTP request to the server.